src/main/sensors/gyro.c

```diff
@@ -280,7 +280,9 @@ void gyroInitFilters(void)
     notchFilter1ApplyFn = nullFilterApply;
     notchFilter2ApplyFn = nullFilterApply;
 
-    if (gyroConfig->gyro_soft_lpf_hz) {  // Initialisation needs to happen once samplingrate is known
+    uint32_t gyroFrequencyNyquist = (1.0f / (gyro.targetLooptime * 0.000001f)) / 2; // No rounding needed
+
+    if (gyroConfig->gyro_soft_lpf_hz && gyroConfig->gyro_soft_lpf_hz <= gyroFrequencyNyquist) {  // Initialisation needs to happen once samplingrate is known
```

Comment:
@borisbstyle , I agree with @mikeller , this is precisely the kind of check that should be done in `validateAndFixConfig`.